modules/apps/29-fee/ibc_module_test.go

```diff
@@ -468,3 +470,301 @@ func (suite *FeeTestSuite) TestOnChanCloseConfirm() {
+		})
+	}
+}
+
+func (suite *FeeTestSuite) TestOnAcknowledgementPacket() {
+	testCases := []struct {
+		name     string
+		malleate func(suite *FeeTestSuite)
+		expPass  bool
+	}{
+		{
+			"success",
+			func(suite *FeeTestSuite) {
+				packetId := channeltypes.NewPacketId(suite.path.EndpointA.ChannelID, suite.path.EndpointA.ChannelConfig.PortID, suite.chainA.SenderAccount.GetSequence())
+				identifiedFee := types.NewIdentifiedPacketFee(
+					packetId,
+					types.Fee{
+						ReceiveFee: validCoins,
+						AckFee:     validCoins2,
+						TimeoutFee: validCoins3,
+					},
+					suite.chainA.SenderAccount.GetAddress().String(),
+					[]string{},
+				)
+				err := suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), identifiedFee)
+				suite.Require().NoError(err)
+			},
+			true,
+		},
+		{
+			"identified packet fee doesn't exist",
+			func(suite *FeeTestSuite) {
+				suite.Require().Error(types.ErrFeeNotFound)
+			},
+			false,
+		},
+		{
+			"refundAccount doesn't exist",
+			func(suite *FeeTestSuite) {
+				packetId := channeltypes.NewPacketId(suite.path.EndpointA.ChannelID, suite.path.EndpointA.ChannelConfig.PortID, suite.chainA.SenderAccount.GetSequence())
+				identifiedFee := types.NewIdentifiedPacketFee(
+					packetId,
+					types.Fee{
+						ReceiveFee: validCoins,
+						AckFee:     validCoins2,
+						TimeoutFee: validCoins3,
+					},
+					suite.chainB.SenderAccount.GetAddress().String(),
+					[]string{},
+				)
+				suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), identifiedFee)
+				suite.Require().Error(types.ErrRefundAccNotFound)
+			},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+			suite.coordinator.SetupConnections(suite.path)
+
+			// open incentivized channel
+			suite.path.EndpointA.ChanOpenInit()
+			suite.path.EndpointB.ChanOpenTry()
+			suite.path.EndpointA.ChanOpenAck()
+			suite.path.EndpointB.ChanOpenConfirm()
+
+			// set up coins
+			coin := ibctesting.TestCoin
+			fungibleTokenPacket := transfertypes.NewFungibleTokenPacketData(
+				coin.Denom,
+				sdk.NewInt(100).Uint64(),
+				suite.chainA.SenderAccount.GetAddress().String(),
+				suite.chainB.SenderAccount.GetAddress().String(),
+			)
+			packet := channeltypes.NewPacket(
+				fungibleTokenPacket.GetBytes(),
+				suite.chainA.SenderAccount.GetSequence(),
+				suite.path.EndpointA.ChannelConfig.PortID,
+				suite.path.EndpointA.ChannelID,
+				suite.path.EndpointB.ChannelConfig.PortID,
+				suite.path.EndpointB.ChannelID,
+				clienttypes.NewHeight(0, 100),
+				0,
+			)
+
+			// set up module and callbacks
+			module, _, err := suite.chainA.App.GetIBCKeeper().PortKeeper.LookupModuleByPort(suite.chainA.GetContext(), ibctesting.TransferPort)
+			suite.Require().NoError(err)
+
+			cbs, ok := suite.chainA.App.GetIBCKeeper().Router.GetRoute(module)
+			suite.Require().True(ok)
+
+			// malleate test case
+			tc.malleate(suite)
```

Comment:
Should we be passing suite into malleate here? This seems unnecessary @colin-axner

Reply:
yea it is unnecessary